### DISCUSSION TITLE
Vampire drain ability

### DISCUSSION THREAD

**Discussion by zenseii** [ihhub/fheroes2#6001] [General]
Hi, @LeHerosInconnu and @Branikolog,

I believe you are the most familiar with the original mechanics and recently I discovered that Vampire Lords actually drain life from zombies. I checked the original game and saw that this is indeed how it works.

I thought maybe they don't from inanimate beings like Elementals, Golems and maybe others, so I would like to check if there is any special rule to it? I haven't checked in fheroes2 any other creature than zombies, which can indeed be drained from like in the OG.

Thanks in advance!

**Comment by LeHerosInconnu** [comment]
Hello @zenseii,

> Hi, @LeHerosInconnu and @Branikolog,
> 
> I believe you are the most familiar with the original mechanics and recently I discovered that Vampire Lords actually drain life from zombies. I checked the original game and saw that this is indeed how it works.
> 
> I thought maybe they don't from inanimate beings like Elementals, Golems and maybe others, so I would like to check if there is any special rule to it? I haven't checked in fheroes2 any other creature than zombies, which can indeed be drained from like in the OG.
> 
> Thanks in advance!

In Heroes 2, Vampire Lords can drain life from all creatures in the game.
I didn't test for all creatures in fheroes2, but it should be the same.

**Reply by zenseii** [reply]
I see! Well, I can take a quick look at the code for fheroes2, but thank you for the clarification about the original game!
